apiserver/common/networkingcommon/linklayer.go

```diff
@@ -209,24 +214,24 @@ func (o *MachineLinkLayerOp) PopulateExistingAddresses() error {
 	return errors.Trace(err)
 }
 
-// MatchingIncoming returns the first incoming interface that matches
-// the input known device, based on name and hardware address.
+// MatchingIncoming returns the first incoming interface
+// that matches the input known device based on name.
 // Nil is returned if there is no match.
 func (o *MachineLinkLayerOp) MatchingIncoming(dev LinkLayerDevice) *network.InterfaceInfo {
-	if matches := o.incoming.GetByNameAndHardwareAddress(dev.Name(), dev.MACAddress()); len(matches) > 0 {
+	if matches := o.incoming.GetByName(dev.Name()); len(matches) > 0 {
 		return &matches[0]
 	}
 	return nil
 }
 
 // MatchingIncomingAddrs finds all the primary addresses on devices matching
-// the input name and hardware address, and returns them as state args.
+// the input name, and returns them as state args.
 // TODO (manadart 2020-07-15): We should investigate making an enhanced
 // core/network address type instead of this state type.
 // It would embed ProviderAddress and could be obtained directly via a method
 // or property of InterfaceInfos.
-func (o *MachineLinkLayerOp) MatchingIncomingAddrs(name, hwAddress string) []state.LinkLayerDeviceAddress {
-	return networkAddressStateArgsForDevice(o.Incoming(), name, hwAddress)
+func (o *MachineLinkLayerOp) MatchingIncomingAddrs(name string) []state.LinkLayerDeviceAddress {
```

Comment:
minor nit: this method name is a bit confusing. Perhaps something like `GetPrimaryDeviceAddresses` would convey its purpose better.

Reply:
I have some changes like this in mind that I want to apply to _develop_ as we iterate it there. I want to ensure 2.8 is fairly static first before starting divergent work.